### DISCUSSION TITLE
How to disable LSP and autocomplete

### DISCUSSION THREAD

**Discussion by souvlakeeb** [LunarVim/LunarVim#4572] [Q&A] [answered]
What can I add to my config to disable LSP and autocomplete on start?

**Comment by frolvanya** [comment]
Here's how to disable `cmp` (autocomplete):
```lua
lvim.builtin.cmp.active = false
```
However, I am not sure how to completely disable LSP (and what's the point of doing it, if autocomplete is already disabled. LSP isn't taking too much resources)

**Reply by souvlakeeb** [reply]
I don't want any code suggestions, errors, or warnings. Not about resources, it's about user experience.

**Reply by frolvanya** [reply]
Then you can check [these](https://www.reddit.com/r/neovim/comments/1ae6iwm/disable_lsp_diagnostics/) options, but to be honest I'd question if I really need Neovim distro at this point, especially when distro is as bloated as Lunarvim

**Comment by kylo252** [comment]
> What can I add to my config to disable LSP and autocomplete on start?

do you mean automatic start? or disable them completely?
the servers are started with an autocmd triggered by using ftplugin files, see: https://www.lunarvim.org/docs/configuration/language-features/language-servers

**Reply by souvlakeeb** [reply]
I honestly don't know the difference, but it sounds like I want to prevent auto start.

**Comment by souvlakeeb** [comment, marked as answer]
```lua
lvim.builtin.autopairs.active = false  -- disable autopairs (smart parentheses)
lvim.builtin.cmp.active = false  -- disable autocomplete
vim.diagnostic.enable(false)  -- disable diagnostics
```
